head_test.go

```diff
@@ -97,44 +97,23 @@ func TestAmendDatapointCausesError(t *testing.T) {
 	}
 
 	app := hb.Appender()
-	_, err = app.Add(labels.Labels{}, 0, 0)
-	if err != nil {
-		t.Fatalf("Failed to add sample: %s", err)
-	}
-	if err = app.Commit(); err != nil {
-		t.Fatalf("Unexpected error committing appender: %s", err)
-	}
+	_, err = app.Add(labels.Labels{}, 1, 0)
+	require.NoError(t, err)
+
+	err = app.Commit()
+	require.NoError(t, err)
 
 	app = hb.Appender()
-	_, err = app.Add(labels.Labels{}, 0, 1)
-	if err != ErrAmendSample {
-		t.Fatalf("Expected error amending sample, got: %s", err)
-	}
-}
 
-func TestDuplicateNaNDatapointNoAmendError(t *testing.T) {
```

Comment:
This test should remain too, as it'll catch if we switch to float comparisons as NaN != NaN